code/obj/item/implant.dm

```diff
@@ -1462,6 +1462,10 @@ ABSTRACT_TYPE(/obj/item/implant/revenge)
 		if(!in_interact_range(M, user))
 			boutput(user, "<span class='alert'>You are too far away from [M]!</span>")
 			return
+		var/datum/mind/U = user.mind
+		if(!U.special_role && (istype(src.imp, /obj/item/implanter/mindhack) || istype(src.imp, /obj/item/implanter/super_mindhack))) // prevents (mostly) crew from using mindhacks against antags, considering it's functionally a de-antager in the hands of crew.
```

Comment:
this won't work under any circumstances because you're checking the `imp` (implant) variable for the implanter's typepath, you should be checking for the implant typepath